app/States/PortfolioState.php

```
@@ -0,0 +1,97 @@
+    /** @var array<string, array{shares: float, cost_basis: float|null}> */
+    public array $positions = [];
+
+    public float $cash = 0.0;
+
+    /**
+     * Calculate current portfolio value from positions and latest prices
+     * This is DERIVED state - calculated on demand, not stored
+     */
+    public function value(): float
+    {
+        $positionValue = collect($this->positions)->sum(function ($position, $ticker) {
```

PHPStan (2.1.31)

Parameter #1 $callback of method Illuminate\Support\Collection<string,array<string, float|null>>::sum() expects (callable(array{shares: float, cost_basis: float|null}): mixed)|string|null, Closure(mixed, mixed): mixed given.
Type #1 from the union: Parameter #2 $ticker of passed callable is required but accepting callable does not have that parameter. It will be called without it.

(argument.type)